src/stack/stack.jl

```diff
@@ -307,24 +307,45 @@ end
     DimStack <: AbstractDimStack
 
     DimStack(data::AbstractDimArray...; kw...)
-    DimStack(data::Tuple{Vararg{AbstractDimArray}}; kw...)
-    DimStack(data::NamedTuple{Keys,Vararg{AbstractDimArray}}; kw...)
-    DimStack(data::NamedTuple, dims::DimTuple; metadata=NoMetadata(); kw...)
+    DimStack(data::Union{AbstractArray,Tuple,NamedTuple}[, dims::DimTuple]; kw...)
 
 DimStack holds multiple objects sharing some dimensions, in a `NamedTuple`.
 
-Notably, their behaviour lies somewhere between a `DimArray` and a `NamedTuple`:
+## Arguments
 
-- indexing with a `Symbol` as in `dimstack[:symbol]` returns a `DimArray` layer.
-- iteration and `map` apply over array layers, as indexed with a `Symbol`.
+- `data`: `AbstractDimArray` or an `AbstractArray`, `Tuple` or `NamedTuple` of `AbstractDimArray`s or `AbstractArray`s.
+- `dims`: `DimTuple` of `Dimension`s. Required when `data` is not `AbstractDimArray`s.
+
+## Keywords
+
+- `name`: `Array` `Tuple` of `Symbol` names for each layer. By default
+    the names of `DimArrays` are used, or `:layer1`, `:layer2`, etc.
+- `metadata`: `AbstractDict` or `NamedTuple` metadata for the stack. 
+- `layersfrom`: A dimension to slice layers from if data is a single
+    `DimArray`. Defaults to `nothing`. 
+
+(These are for advanced uses)
+- `layerdims`: `Array`, `Tuple` or `NamedTuple` of dimension tuples to match the
+    dimensions of each layer. Dimensions in `layerdims` must also be in `dims`.
+- `layermetadata`: `Array`, `Tuple` or `NamedTuple` of metadata for each layer.
+- `refdims`: `NamedTuple` of `Dimension`s for each layer, `()` by default.
+
+## Details
+
+`DimStack` behaviour lies somewhere between a `DimArray` and a `NamedTuple`:
+
+- indexing with a `Symbol` as in `dimstack[:layername]` or using `getproperty` 
+    `dimstack.layername` returns a `DimArray` layer.
 - `getindex` or `view` with `Int`, `Dimension`s or `Selector`s that resolve to `Int` will
     return a `NamedTuple` of values from each layer in the stack.
-    This has very good performance, and avoids the need to always use `map`.
 - `getindex` or `view` with a `Vector` or `Colon` will return another `DimStack` where
     all data layers have been sliced.
```

Comment:
```suggestion
- `getindex` or `view` with a `Vector` or `Colon` will return another `DimStack` where
    all data layers have been sliced, unless this resolves to a single element, in which case 
    `getindex` returns a `NamedTuple`
```
Still a bit clunky maybe